core/admin/port_management.py

```
@@ -0,0 +1,826 @@
+            "-d", container_ip,
+            "--dport", str(internal_port),
+            "-j", "ACCEPT"
+        ], check=True, capture_output=True)
+        
+        logger.info(f"Added iptables rules for port {external_port} -> {container_ip}:{internal_port}")
+    except subprocess.CalledProcessError as e:
+        logger.exception(f"Failed to add iptables rules: {e}")
+        raise ValueError(f"Failed to open port: {e}") from e
+    
+    # Calculate times
+    now = datetime.utcnow()
```

Ruff (0.14.11)

`datetime.datetime.utcnow()` used

(DTZ003)